File: app/src/main/java/com/travlog/android/apps/services/ApiException.java
package com.travlog.android.apps.services;

import android.support.annotation.NonNull;

import com.travlog.android.apps.services.apiresponses.Envelope;

/**
 * An exception class wrapping an {@link Envelope}.
 */
public final class ApiException extends ResponseException {
    private final Envelope errorEnvelope;

    public ApiException(final @NonNull Envelope errorEnvelope, final @NonNull retrofit2.Response response) {
        super(response);
        this.errorEnvelope = errorEnvelope;
    }

    public @NonNull
    Envelope errorEnvelope() {
        return this.errorEnvelope;
    }
}

File: app/src/main/java/com/travlog/android/apps/services/ApiException.kt
package com.travlog.android.apps.services

import com.travlog.android.apps.services.apiresponses.Envelope

/**
 * An exception class wrapping an [Envelope].
 */
class ApiException(private val errorEnvelope: Envelope, response: retrofit2.Response<*>) : ResponseException(response) {

    fun errorEnvelope(): Envelope {
        return this.errorEnvelope
    }
}
